### PR TITLE
Extend oidc provider options for WASM 

### PR DESCRIPTION
**Pull request by nahidf**
Adding `AdditionalProviderParameters ` property to oidc provider options, this helps to pass extra parameters to oidc provider on authorization flow 

Addresses https://github.com/dotnet/aspnetcore/issues/25004

### CODE REVIEW

**Comment by SteveSandersonMS** [comment]
Thanks @nahidf!

@javiercn @captainsafia Could you comment on the applicability of this?

File: src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs [hunk 1]
```diff
@@ -57,5 +57,10 @@ public class OidcProviderOptions
         /// </summary>
         [JsonPropertyName("response_mode")]
         public string ResponseMode { get; set; }
+
+        /// <summary>
+        /// Gets or sets the extra query parameters to use on the authorization flow. 
+        /// </summary>
+        public IDictionary<string, string> ExtraQueryParams { get; } = new Dictionary<string, string>();
```
Comment (javiercn) [inline src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs:64]: Change the name to AdditionalProviderParameters or something more descriptive. We normally don't abbreviate these things.

**Review by javiercn — CHANGES_REQUESTED** [review]
Looks good to me in general, but we need to change the property name and add remarks about its usage.

File: src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs (hunk 1 above)
Comment (nahidf) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs:64]: Do you have any suggestion for the json prop name as well? after rename default would be `additionalProviderParameters` in this case @javiercn

**Review by captainsafia — APPROVED** [review]
LGTM accounting for Javier's feedback!

File: src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs [hunk 2]
```diff
@@ -59,8 +59,11 @@ public class OidcProviderOptions
...
 
         /// <summary>
-        /// Gets or sets the extra query parameters to use on the authorization flow. 
+        /// Gets or sets the additional provider parameters to use on the authorization flow. 
         /// </summary>
-        public IDictionary<string, string> ExtraQueryParams { get; } = new Dictionary<string, string>();
+        /// <remarks>
+        /// These parameters are for the IdP and not for the application. Using those parameters in the application in any way on the login callback will likely introduce security issues as they should be treated as untrusted input.
+        /// </remarks>
+        public IDictionary<string, string> AdditionalProviderParameters { get; } = new Dictionary<string, string>();
```
Comment (javiercn) [inline src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs:67]:
```suggestion
        [JsonPropertyName("extraQueryParams")]
        public IDictionary<string, string> AdditionalProviderParameters { get; } = new Dictionary<string, string>();
```

Comment (campersau) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs:68]: All other json property names are using underscores and lowercase. Should this be `extra_query_params` to be consistent or is this intentional?

Comment (javiercn) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs:68]: This is intentional to match the UserManagerSettings object of oidc-client JS shape.

Comment (nahidf) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/Options/OidcProviderOptions.cs:68]: @javiercn would you please re-review this PR?

**Review by javiercn — APPROVED** [review]

**Comment by Vizual-Dev** [comment]
Any idea when this will make it into an official release?

**Comment by captainsafia** [comment]
@Vizual-Dev This is shipped as part of .NET 6 Preview 1.

**Comment by vanillajonathan** [comment]
This does not cover all use cases, it is static and cannot be used to dynamically add query parameters.

Example:
```cs
OnRedirectToIdentityProvider = (context) =>
{
    var tenant = context.Request.Query.FirstOrDefault(x => x.Key == "tenant");
    context.ProtocolMessage.SetParameter("tenant", tenant);
    return Task.CompletedTask;
}
```

**Pull request merged**